src/test/integration/users/AVS.t.sol

```diff
@@ -0,0 +1,199 @@
+    using SingleItemArrayLib for *;
+
+    Vm constant cheats = Vm(VM_ADDRESS);
+    AllocationManager immutable allocationManager;
+    StrategyFactory immutable strategyFactory;
+    TimeMachine immutable timeMachine;
+    string _NAME;
+
+    constructor(
+        string memory name
+    ) {
+        IAVSDeployer deployer = IAVSDeployer(msg.sender);
+        allocationManager = deployer.allocationManager();
+        strategyFactory = deployer.strategyFactory();
+        timeMachine = deployer.timeMachine();
+        _NAME = name;
+    }
+
+    modifier createSnapshot() virtual {
+        timeMachine.createSnapshot();
+        _;
+    }
+
+    receive() external payable {}
+
+    function NAME() public view override returns (string memory) {
+        return _NAME;
+    }
+
+    /// -----------------------------------------------------------------------
+    /// AllocationManager Interactions
+    /// -----------------------------------------------------------------------
+    
+    function createOperatorSets(
+        uint256 numOpSets,
+        uint256 numStrategies
+    ) public createSnapshot returns (uint32[] memory operatorSetIds, IStrategy[][] memory strategies) {
+        _logM("createOperatorSets");
+
+        CreateSetParams[] memory p = new CreateSetParams[](numOpSets);
+        operatorSetIds = new uint32[](numOpSets);
+        strategies = new IStrategy[][](numOpSets);
+
+        for (uint256 i; i < numOpSets; ++i) {
+            p[i].operatorSetId = uint32(cheats.randomUint({bits: 32}));
+            for (uint256 j; j < numStrategies; ++j) {
+                IERC20 token = IERC20(new ERC20Mock());
+                p[i].strategies[j] = strategyFactory.deployNewStrategy(token);
+            }
+
+            operatorSetIds[i] = p[i].operatorSetId;
+            strategies[i] = p[i].strategies;
+        }
+
+        _logCreateOperatorSets(p);
+
+        allocationManager.createOperatorSets(p);
+    }
+
+    function createOperatorSet(IStrategy[] memory strategies) public createSnapshot returns (uint32 operatorSetId) {
+        _logM("createOperatorSets");
+
+        CreateSetParams[] memory p = CreateSetParams({
+            operatorSetId: operatorSetId,
+            strategies: strategies
+        }).toArray();
+
+        _logCreateOperatorSets(p);
+
+        allocationManager.createOperatorSets(p);
+    }
+
+    function slashOperator(User operator, uint32 operatorSetId, uint256 wadToSlash) public createSnapshot {
+        _logM("slashOperator");
+
+        SlashingParams memory p = SlashingParams({
+            operator: address(operator),
+            operatorSetId: operatorSetId,
+            wadToSlash: wadToSlash,
+            description: "bad operator"
+        });
+
+        _logSlashOperator(p);
+
+        allocationManager.slashOperator(p);
+    }
+
+    function slashOperator(User operator, uint32 operatorSetId) public {
+        slashOperator(operator, operatorSetId, cheats.randomUint(0.1 ether, 1 ether));
```

Comment:
Lets keep randomization out of the avs contract, so just the previous slashOperator functionality is fine